components/org.wso2.carbon.identity.idle.account.identification/src/main/java/org/wso2/carbon/identity/idle/account/identification/services/IdleAccountIdentificationServiceImpl.java

```diff
@@ -0,0 +1,91 @@
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+
+/**
+ * Implementation of the service interface for idle account identification.
+ */
+public class IdleAccountIdentificationServiceImpl implements IdleAccountIdentificationService {
+
+    IdleAccIdentificationDAO idleAccIdentificationDAO = new IdleAccIdentificationDAOImpl();
+
+    @Override
+    public List<InactiveUserModel> getInactiveUsers(String inactiveAfter, String excludeBefore, String tenantDomain)
+            throws IdleAccIdentificationException {
+
+        validateDates(inactiveAfter, excludeBefore);
+        return idleAccIdentificationDAO.getInactiveUsers(inactiveAfter, excludeBefore, tenantDomain);
+    }
+
+    /**
+     * Validate the dates.
+     *
+     * @param inactiveAfter date after which the user should be inactive
+     * @param excludeBefore date before which the user should be inactive
+     * @throws IdleAccIdentificationClientException Exception thrown when the dates are invalid
+     */
+    private void validateDates(String inactiveAfter, String excludeBefore) throws IdleAccIdentificationClientException {
+
+        // Check if the required parameters are present.
+        if (StringUtils.isEmpty(inactiveAfter)) {
+            throw new IdleAccIdentificationClientException(
+                    ErrorMessages.ERROR_REQUIRED_PARAMETER_MISSING.getCode(),
+                    ErrorMessages.ERROR_REQUIRED_PARAMETER_MISSING.getMessage(),
+                    String.format(ErrorMessages.ERROR_REQUIRED_PARAMETER_MISSING.getDescription(),
+                            IdleAccIdentificationConstants.DATE_INACTIVE_AFTER));
+        }
+
+        // Check if the date format is valid.
+        Utils.validateDateFormat(inactiveAfter, excludeBefore);
+
+        try {
+            LocalDateTime inactiveAfterObj = LocalDate.parse(inactiveAfter).atStartOfDay();
+            if (StringUtils.isNotEmpty(excludeBefore)) {
+                LocalDateTime excludeBeforeObj = LocalDate.parse(excludeBefore).atStartOfDay();
+            }
+        } catch (DateTimeParseException e) {
+            if (inactiveAfter.equals(e.getParsedString())) {
+                throw new IdleAccIdentificationClientException(
+                        ErrorMessages.ERROR_INVALID_DATE.getCode(), ErrorMessages.ERROR_INVALID_DATE.getMessage(),
+                        String.format(ErrorMessages.ERROR_INVALID_DATE.getDescription(),
+                                IdleAccIdentificationConstants.DATE_INACTIVE_AFTER));
+            } else {
+                throw new IdleAccIdentificationClientException(
+                        ErrorMessages.ERROR_INVALID_DATE.getCode(), ErrorMessages.ERROR_INVALID_DATE.getMessage(),
+                        String.format(ErrorMessages.ERROR_INVALID_DATE.getDescription(),
+                                IdleAccIdentificationConstants.DATE_EXCLUDE_BEFORE));
+            }
```

Comment:
```suggestion
            } 
            throw new IdleAccIdentificationClientException(
                        ErrorMessages.ERROR_INVALID_DATE.getCode(), ErrorMessages.ERROR_INVALID_DATE.getMessage(),
                        String.format(ErrorMessages.ERROR_INVALID_DATE.getDescription(),
                                IdleAccIdentificationConstants.DATE_EXCLUDE_BEFORE));
            
```

Reply:
Fixed in https://github.com/wso2-extensions/identity-governance/pull/698/commits/891e0922ab1f1a685f3049500ff4ab014bdec860